### ISSUE TITLE
Gradient for tf.sparse.reduce_max

### ISSUE THREAD

**Issue by AmadeusloveIris** [tensorflow/tensorflow#54214]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**Describe the feature and the current behavior/state.**
tf.sparse.reduce_max do not define grad. So it cannot be used in training step.

**Will this change the current api? How?**
Won't change current api. Just add grad in tf.sparse.reduce_max

**Who will benefit with this feature?**
People using sparse.reduce_max in training. If they have huge sparse matrix and cannot turn to dense array.

**Any Other info.**
I am working with pointnet-like model. And it will take a huge sparse matrix. If I turn this sparse matrix to dense, it will take ~60G memory. So I need do global maxpooling. Using tf sparse reduce_max. Unfortunately, it cannot be used in training right now.
Thanks

**Comment by sushreebarsa** [comment]
@AmadeusloveIris 
Can you please elaborate about your feature and please specify the Use Cases for this feature. Thanks!
